queries/parse.go

```diff
@@ -74,52 +77,194 @@ func AppendMatcher(queryValues url.Values, queryValuesForAuth url.Values, key st
 
-	if cluster != "" {
-		authResourceNames = append(authResourceNames, fmt.Sprintf("%s-%s-%s-%s", hubKey, hub, clusterKey, cluster))
-		authScopeNames = append(authScopeNames, "GET")
+		for _, matcherSelector := range matcher {
+			switch matcherSelector.Name {
+			case "cluster":
+				clusterMatcher = matcherSelector
+			case "namespace":
+				namespaceMatcher = matcherSelector
+			case "exported_namespace":
+				exportedNamespaceMatcher = matcherSelector
+			default:
+				continue
+			}
+		}
 
-		exported_namespace, _ := AppendMatcher(queryValues, queryValuesForAuth, "exported_namespace", fmt.Sprintf("%s-%s-%s-%s-%s", hubKey, hub, clusterKey, cluster, projectKey), "")
-		namespace, _ := AppendMatcher(queryValues, queryValuesForAuth, "namespace", fmt.Sprintf("%s-%s-%s-%s-%s", hubKey, hub, clusterKey, cluster, projectKey), exported_namespace)
+		if clusterMatcher == nil || clusterMatcher.Value == "" {
+			continue
+		}
 
-		if namespace != "" {
-			if cluster != "" {
-				authResourceNames = append(authResourceNames, fmt.Sprintf("%s-%s-%s-%s-%s-%s", hubKey, hub, clusterKey, cluster, projectKey, namespace))
-				authScopeNames = append(authScopeNames, "GET")
+		var clusterValues []string
+		if clusterMatcher.Type == labels.MatchEqual {
+			clusterValues = append(clusterValues, clusterMatcher.Value)
+		} else {
+			clusterValues = strings.Split(clusterMatcher.Value, "|")
+			for j, clusterValue := range clusterValues {
+				clusterValues[j] = regexp.QuoteMeta(clusterValue)
 			}
 		}
+
+		var namespaceValues []string
+		if exportedNamespaceMatcher != nil {
+			if exportedNamespaceMatcher.Type == labels.MatchEqual {
+				namespaceValues = append(namespaceValues, exportedNamespaceMatcher.Value)
+			} else {
+				namespaces := strings.Split(exportedNamespaceMatcher.Value, "|")
+				for _, namespace := range namespaces {
+					namespaceValues = append(namespaceValues, regexp.QuoteMeta(namespace))
+				}
+			}
+		}
+		if namespaceMatcher != nil {
+			if namespaceMatcher.Type == labels.MatchEqual {
+				namespaceValues = append(namespaceValues, namespaceMatcher.Value)
+			} else {
+				namespaces := strings.Split(namespaceMatcher.Value, "|")
+				for _, namespace := range namespaces {
+					namespaceValues = append(namespaceValues, regexp.QuoteMeta(namespace))
+				}
+			}
+		}
+
+		if len(namespaceValues) == 0 {
+			var clusterResources []string
+			for _, clusterValue := range clusterValues {
+				clusterResources = append(clusterResources, fmt.Sprintf("%s-%s-%s-%s", hubKey, hub, clusterKey, clusterValue))
+				uniqueResources[fmt.Sprintf("%s-%s-%s-%s", hubKey, hub, clusterKey, clusterValue)] = struct{}{}
+			}
+			resources[i] = append(resources[i], clusterResources)
+			continue
+		}
+
+		/* BEGIN assisted by Claude */
+		totalCombinations := 1 << len(clusterValues)
+		for j := 0; j < totalCombinations; j++ {
+			var namespaceResources []string
+			for k, clusterValue := range clusterValues {
+				if (j>>k)&1 == 0 {
```

Comment:
@ajamias can you please explain the bit shifting and variables here a little more? Other than that, this code works amazing!